Guide: Using Open Safety Models (#66)
roostorg/model-community: resources/RMC Guide to Using Open Safety Models.md

> # A ROOST Model Community Guide to Using Open Safety Models

Reviewer: Nit: we could shorten this a bit, plus use it as an opportunity to link back the the RMC (in case it's shared elsewhere)

```suggestion
# Using Open Safety Models

A guide from the [ROOST Model Community](https://github.com/roostorg/model-community)
```

## Setting the Stage: Open AI Models in the Modern T&S Landscape

> 
> Modern trust & safety (T&S) teams require solutions to four functional archetypes, as outlined by the [DIRE framework](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=5369158):
> 
> - **Detection:** Identifying potential risks in accounts, behaviors, and content through behavioral signals and existing databases (such as via hash matching, which identifies known threats based on their digital fingerprint). This is how you identify when something has gone wrong.
> - **Investigation:** Analyzing broad attack patterns by evaluating context beyond individual entities, or diving deep into a single incident. This is how you understand why exactly things have gone wrong.
> - **Review:** Assessing content against policies to determine appropriate next steps. This is how you determine the best course of action in nuanced cases.
> - **Enforcement:** Taking actions and meeting reporting obligations. This is how you remove the threat from your platform.

Reviewer: Protip: separating longer bullet points with a newline will make them a bit easier to read by wrapping each in a paragraph

```suggestion
- **Detection:** Identifying potential risks in accounts, behaviors, and content through behavioral signals and existing databases (such as via hash matching, which identifies known threats based on their digital fingerprint). This is how you identify when something has gone wrong.

- **Investigation:** Analyzing broad attack patterns by evaluating context beyond individual entities, or diving deep into a single incident. This is how you understand why exactly things have gone wrong.

- **Review:** Assessing content against policies to determine appropriate next steps. This is how you determine the best course of action in nuanced cases.

- **Enforcement:** Taking actions and meeting reporting obligations. This is how you remove the threat from your platform.
```

AI models fine-tuned for safety can support teams across all four quadrants, but they're most well-suited to Detection and Review. Historically, detection tooling has been broadly commercially available. However, existing tools have failed on two major fronts. They are often (1) highly manual or reliant on user reporting and (2) fragmented or hard to integrate across internal tools and vendors.

> We define open safety models as models that are free to access, available to deploy agnostic of platform, and finetuned specifically for a T&S use case – you can read more in the [ROOST Model Community (RMC) ReadMe](https://github.com/roostorg/model-community). These models offer a solution to our two main historic problems. For example: smaller developers may lack the resources necessary to run high-end private models; those who deal with sensitive personally identifiable information may want to keep their data private; and many teams may prefer to use their own policies or further fine-tune specific models on their policies.

Reviewer: Just a few small stylistic suggestions

```suggestion
We define open safety models as models that are free to access, available to deploy agnostic of platform, and fine-tuned specifically for a T&S use case; you can read more in the [ROOST Model Community (RMC) README](https://github.com/roostorg/model-community). These models offer a solution to our two main historic problems. For example: smaller developers may lack the resources necessary to run high-end private models; those who deal with sensitive personally identifiable information may want to keep their data private; and many teams may prefer to use their own policies or further fine-tune specific models on their policies.
```

---

## Using Open Models in Your T&S Architecture

There are three particularly common use cases for modern open models in a Trust & Safety setting, although this is far from an exhaustive list:

1. **Content Detection:** You can use models to classify content according to an existing policy, which has historically been the typical use case for classifiers and often results in a binary decision (e.g., "policy violation" vs. "acceptable").
2. **Investigation & Review Support:** You can use models to support your investigation and review operations (whether automated or human-led), especially with reasoning models that can better handle abstracted or pattern-matched threats.
  - *Case Study:* Check out [Phoebe](https://github.com/haileyok/phoebe), an agentic solution that automates the entire investigation process by analyzing threat patterns and writing new rules to stop those threats via ROOST's Osprey engine.
3. **Iterative Improvement:** You can use models to iterate on your existing policies and processes, especially as you work to build policies that are increasingly interpretable by both human safety teams and AI.

>   - *Case Study:* Participants at the Open Safeguard Hackathon built a [Policy Agent CLI](https://github.com/roostorg/model-community/tree/main/gpt-oss-safeguard/projects/open-safeguard-dec-2025/policy-agent-cli) to test, improve, and iterate on the performance of specific policies.

Reviewer: Same whitespacce suggestion as above:

```suggestion
1. **Content Detection:** You can use models to classify content according to an existing policy, which has historically been the typical use case for classifiers and often results in a binary decision (e.g., "policy violation" vs. "acceptable").

2. **Investigation & Review Support:** You can use models to support your investigation and review operations (whether automated or human-led), especially with reasoning models that can better handle abstracted or pattern-matched threats.

   - *Case Study:* Check out [Phoebe](https://github.com/haileyok/phoebe), an agentic solution that accelerates the entire investigation process by analyzing threat patterns and suggesting new rules to stop those threats via ROOST's Osprey engine.

3. **Iterative Improvement:** You can use models to iterate on your existing policies and processes, especially as you work to build policies that are increasingly interpretable by both human safety teams and AI.

   - *Case Study:* Participants at the Open Safeguard Hackathon built a [Policy Agent CLI](https://github.com/roostorg/model-community/tree/main/gpt-oss-safeguard/projects/open-safeguard-dec-2025/policy-agent-cli) to test, improve, and iterate on the performance of specific policies.
```

I also wonder about describing Phoebe as automating "the entire investigation process." I think it would be more accurate to say it _accelerates_ the investigation process by analyzing threat patterns and _suggesting_ new rules to stop those threats, since my understanding is that it is not fully autonomous but is directly driven by a human. I've suggested that phrasing above. I could be wrong though, and I guess I can tag @haileyok to ask for her thoughts on the phrasing 😄

Author: I like the wording edit made here for now...good flag though

These three use cases are an oversimplification. They are also not mutually exclusive (and in fact often intersect). For example, platforms are increasingly considering a two-pronged approach to using safety models. The first model, usually smaller and lower latency, is a first layer of defense that renders judgments on easy to classify cases. Any edge cases that raise uncertainty can then be routed to the second model, often a higher performance model with reasoning capabilities, which proposes a recommended action for human review. The data from those decisions can (and should) then be used to iterate on policies.

> Most importantly, these use cases are only the surface of open safety models' capabilities. Newer, more capable models unlock the possibility that T&S teams can automate more complex tasks, such as clustering content for deeper investigations, parsing multi-turn conversations for hidden threats (e.g., grooming), or the use of agentic moderators. These more advanced use cases are particularly valuable as your platform scales – they can help with account level evaluations and broader safety considerations (e.g., proactively stopping fraud campaigns).

Reviewer: grammar nit: parallelism

```suggestion
Most importantly, these use cases are only the surface of open safety models' capabilities. Newer, more capable models unlock the possibility that T&S teams can automate more complex tasks, such as clustering content for deeper investigations, parsing multi-turn conversations for hidden threats (e.g., grooming), or using agentic moderators. These more advanced use cases are particularly valuable as your platform scales – they can help with account level evaluations and broader safety considerations (e.g., proactively stopping fraud campaigns).
```

- **Determine whether you want to run the model locally or via hosted inference API.** Local models afford full data privacy control and configurability, while APIs are often easier to set up.
  - This decision affects other considerations, such as latency – running the model locally on a Mac without a cloud GPU will likely lead to slower times than using an inference provider or setting up that GPU.
- **Test and refine the model before deploying.** Make sure that your model is performing sufficiently well on your policies before deploying it! To do this, you may want to:
  - Use or build a structured, high quality "golden" dataset to compare model performance against. This can often be a time and cost intensive practice, so don't be afraid to ask RMC peers if they have datasets to share!
  - Identify what metrics you are most worried about (e.g., precision vs. recall).
  - Understand whether errors stem from model performance or policy inconsistency.
  - Consider further model fine-tuning (if cost and time allow). Because out of the box classifiers are not customized to your specific policies, they may not always perform as you'd like without additional fine-tuning.

> - For developers worried about managing costs, consider starting your T&S workflows with smaller, more lightweight models. Many basic, high volume tasks can be accomplished quickly and efficiently by smaller models. Consider where you really need to use a larger, more complex model. Validate your approach by running candidate models on sample batches of data before committing to a full deployment, and adjust your approach based on the results you find.

Reviewer: Same readability tip as above :)

```suggestion
- **Decide which model to use.** This should factor in questions such as the model's license (are there any restrictions that are prohibitive to you?), the model's size (will its latency be sufficient?), and the model's performance (often indicated via a model card or evaluation results).

- **Determine whether you want to run the model locally or via hosted inference API.** Local models afford full data privacy control and configurability, while APIs are often easier to set up.

  - This decision affects other considerations, such as latency – running the model locally on a Mac without a cloud GPU will likely lead to slower times than using an inference provider or setting up that GPU.

- **Test and refine the model before deploying.** Make sure that your model is performing sufficiently well on your policies before deploying it! To do this, you may want to:

  - Use or build a structured, high quality "golden" dataset to compare model performance against. This can often be a time and cost intensive practice, so don't be afraid to ask RMC peers if they have datasets to share!

  - Identify what metrics you are most worried about (e.g., precision vs. recall).

  - Understand whether errors stem from model performance or policy inconsistency.

  - Consider further model fine-tuning (if cost and time allow). Because out of the box classifiers are not customized to your specific policies, they may not always perform as you'd like without additional fine-tuning.

For developers worried about managing costs, consider starting your T&S workflows with smaller, more lightweight models. Many basic, high volume tasks can be accomplished quickly and efficiently by smaller models. Consider where you really need to use a larger, more complex model. Validate your approach by running candidate models on sample batches of data before committing to a full deployment, and adjust your approach based on the results you find.
```
